PyLeap/Views/Project Views/ProjectViewModel.swift

```diff
@@ -38,4530 +35,351 @@ class ProjectViewModel: ObservableObject  {
     
     @Published var isConnectedToInternet = false
     
-    let directoryPath = FileManager.default.urls(for: .documentDirectory, in: .userDomainMask)[0]
-    let cachesPath = FileManager.default.urls(for: .cachesDirectory, in: .userDomainMask)[0]
-    
-    @Published var newBundleDownloaded = false
     @Published var didCompleteTranfer = false
     @Published var writeError = false
     
-    
-    func completedTransfer() {
-        
-        didCompleteTranfer = true
-        
-        DispatchQueue.main.asyncAfter(deadline: .now() + 3) {
-            self.didCompleteTranfer = false
-            
-        }
-    }
-    
-    func displayErrorMessage() {
-        DispatchQueue.main.async {
-            self.writeError = true
-            self.sendingBundle = false
-        }
-        
-        DispatchQueue.main.asyncAfter(deadline: .now() + 10) {
-            self.writeError = false
-            
-        }
-        
-    }
-    
-    func downloadCheck(at filePath: URL){
-       
-        DispatchQueue.main.asyncAfter(deadline: .now() + 0.5) {
-            print("Download Check!")
-            do {
-                
-                if FileManager.default.fileExists(atPath: filePath.path) {
-                    self.didDownload = true
-                    
-                    
-                    DispatchQueue.main.async {
-                        print("FILES AVAILABLE")
-                        self.didDownload = false
-                        self.didDownload = true
-                        self.filesDownloaded(url: filePath)
-                        
-                    }
-                } else {
-                    DispatchQueue.main.async {
-                        print("FILES NOT DOWNLOADED")
-                       
-                        
-                    }
-                }
-            } catch {
-                print(error)
-            }
-        }
-       
-        
-        
-    }
-    
-    func internetMonitoring() {
-
-        networkMonitor.startMonitoring()
-        
-            networkMonitor.monitor.pathUpdateHandler = { path in
-            if path.status == .satisfied {
-                print("We're connected!")
-                
-                DispatchQueue.main.async {
-                    self.showAlert = false
-                    self.isConnectedToInternet = true
-                }
-                
-                
-                
-            } else {
-                print("No connection.")
-                DispatchQueue.main.async {
-                    self.showAlert = true
-                    self.isConnectedToInternet = false
-                }
-                
-            }
-
-            print("isExpensive :\(path.isExpensive)")
-        }
-        
-    }
+    var projectDirectories: [URL] = []
```

Comment:
Please add a comment at the top of the file here that explains how files are transfered. For example, include how the different functions are related to each other.